### PR TITLE
Added HTML parsing to XML module, references #1240

### PR DESCRIPTION
**Pull request by ryanworl**
Let me know if you'd like more tests. I think since the HTML parsing functions of libxml2 return the same document structure as the XML ones it isn't worth duplicating most of the tests that already exist.

### CODE REVIEW

File: src/xml/xml.cr [hunk 1]
```diff
@@ -19,6 +19,10 @@ module XML
       )
   end
 
+  def self.parseHTML(string: String, options = HTMLParserOptions.default : HTMLParserOptions)
```
Comment (jhass) [inline src/xml/xml.cr]: We're using snake_case for method names, it should be `parse_html`

**Comment by jhass** [comment]
Not to be done here, but something to think about in the mid-term if we're continuing to add HTML related functionality to the XML class is to rename it. The superset of XML and HTML would be SGML, though that is less well known, other candidates that quickly come to mind are Document or DOM.

**Comment by bararchy** [comment]
body p { margin-bottom: 0cm; margin-top: 0pt; } 
  
  
    
    
    On 08/20/2015 02:53 PM, Jonne Haß
      wrote:
    
    
      
      Not to be done here, but something to think about in the
        mid-term if we're continuing to add HTML related functionality
        to the XML class is to rename it. The superset of XML and HTML
        would be SGML, though that is less well known, other candidates
        that quickly come to mind are Document or DOM.
      —
        Reply to this email directly or view
          it on GitHub.
      
        
          
          
          
          
        
        
        
      
    
    
    +1 for Document::XML and Document::HTML

**Comment by ryanworl** [comment]
:+1: for Document from me.

**Comment by ryanworl** [comment]
Are there any issues with this PR that I need to fix before it is merged?

**Comment by jhass** [comment]
It no longer cleanly applies, so a rebase couldn't hurt, but I think it's just nobody with commit rights had the time to review it yet ;)

**Comment by asterite** [comment]
@ryanworl Your PR is perfect, it's just that right now I have time to merge it. I'll merge with the command line. I'll also try to add a way to parse an HTML from an IO.

**Comment by asterite** [comment]
@ryanworl Thanks!! :heart::yellow_heart::blue_heart::purple_heart::green_heart:

**Comment by asterite** [comment]
Maybe `Document` is too general, I actually don't mind the XML namespace... and then HTML is kind of like XML, so...

**Comment by jhass** [comment]
Ew, it really is not, sorry.

**Pull request merged**